core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -346,6 +340,52 @@ public void accept(Path path) {
         }
     }
 
+    /**
+     * Manifest generation is quite simple : we just have to push some attributes in manfest.
+     * However, it gets a little more complex is manifest preexists.
+     * So we first try to see if a manifest exists, and otherwise create a new one.
+     * 
+     * <b>BEWARE</b> this method should be invoked after file copy from target/classes and so on.
+     * Otherwise this manifest manipulation will be useless.
+     * 
+     * @see https://github.com/quarkusio/quarkus/issues/1443
+     * @param runnerZipFs
+     * @param classPath
+     * @throws IOException
+     */
+    private void generateManifest(FileSystem runnerZipFs, final String classPath) throws IOException {
+        final Path manifestPath = runnerZipFs.getPath("META-INF", "MANIFEST.MF");
+        final Manifest manifest = new Manifest();
+        if (Files.exists(manifestPath)) {
+            log.warn(
+                    "Seems like you have a pre-existing MANIFEST.MF. Since quarkus generates an executable JAR, some entries will be added.");
+            try (InputStream is = Files.newInputStream(manifestPath)) {
+                manifest.read(is);
+            }
+        }
+        Attributes attributes = manifest.getMainAttributes();
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        if (attributes.containsKey(Attributes.Name.CLASS_PATH)) {
+            log.warn(
+                    "Your MANIFEST.MF defines a CLASSPATH. We don't yet support merge of CLASSPATHS. So your CLASSPATH will be overwritten by Quarkus one.");
+        }
+        attributes.put(Attributes.Name.CLASS_PATH, classPath);
+        if (attributes.containsKey(Attributes.Name.MAIN_CLASS)) {
+            String existingMainClass = attributes.getValue(Attributes.Name.MAIN_CLASS);
+            if (!mainClass.equals(existingMainClass)) {
+                log.errorf("Your MANIFEST.MF defines as %s %s, while Quarkus is configured to use %s. "
```

Comment:
Maybe "Your MANIFEST.MF already defined a MAIN_CLASS entry. Quarkus has overwritten your existing entry." would be better.

You can have the name of the entries as a variable but in this case, please be consistent across the 2 error message.